src/jit/hwintrinsicxarch.cpp

```diff
@@ -117,29 +120,40 @@ InstructionSet Compiler::lookupHWIntrinsicISA(const char* className)
 }
 
 //------------------------------------------------------------------------
-// lookupHWIntrinsic: map intrinsic name to named intrinsic value
+// lookupHWIntrinsic: map intrinsic name to named intrinsic value for
+//                    hardware or RyuJIT implemented intrinsics or return
+//                    NI_Illegal for software implemented intrinsics for
+//                    further compilation and expected inlining
 //
 // Arguments:
 //    methodName -- name of the intrinsic function.
 //    isa        -- instruction set of the intrinsic.
 //
 // Return Value:
-//    Id for the hardware intrinsic.
+//    Id for the hardware intrinsic or NI_Illegal for software intrinsics.
 //
 // TODO-Throughput: replace sequential search by binary search
 NamedIntrinsic Compiler::lookupHWIntrinsic(const char* methodName, InstructionSet isa)
 {
     NamedIntrinsic result = NI_Illegal;
     if (isa != InstructionSet_ILLEGAL)
     {
-        for (int i = 0; i < NI_HW_INTRINSIC_END - NI_HW_INTRINSIC_START - 1; i++)
+        for (int i = 0; i < NI_SW_INTRINSIC_END - NI_HW_INTRINSIC_START - 1; i++)
         {
             if (isa == hwIntrinsicInfoArray[i].isa && strcmp(methodName, hwIntrinsicInfoArray[i].intrinsicName) == 0)
             {
                 result = hwIntrinsicInfoArray[i].intrinsicID;
             }
         }
+
+        if ((result > NI_SW_INTRINSIC_START) && (result < NI_SW_INTRINSIC_END))
+        {
+            return NI_Illegal;
+        }
```

Comment:
This design changes the framework too much. I prefer to use `HARDWARE_INTRINSIC` with `HW_Flag_NoCodeGen` in the original table.

Reply:
Yeah, I wonder if defining a new group is the right thing to be doing.

It seems like a flag would be sufficient. Although we may want something that differentiates `NoCodeGen` and `Implemented in Managed Code`.

Reply:
Agree, we can change the name of `HW_Flag_NoCodeGen`.